src/app/CommandSender.cpp

```diff
@@ -330,5 +355,62 @@ TLV::TLVWriter * CommandSender::GetCommandDataIBTLVWriter()
     }
 }
 
+CHIP_ERROR CommandSender::SendTimedRequest(uint16_t aTimeoutMs)
```

Comment:
maybe we can have TimedRequest.cpp/h for this behavior, then have Send function, and Process function, which can be reused by Command and Write
Similar to StatusResponse.cpp/h

Reply:
I am planning to work on Write next (tomorrow, in fact), and I will absolutely factor out any code that ends up shareable between the two.  But I wanted to stand up invoke first.

Reply:
Sounds good, thanks